securesystemslib/interface.py

```diff
@@ -900,6 +900,260 @@ def import_ecdsa_privatekey_from_file(filepath, password=None):
   return key_object
 
 
+def generate_and_write_spx_keypair(filepath=None, password=None):
+  """
+  <Purpose>
+    Generate an SPX keypair, where the encrypted key (using 'password' as
+    the passphrase) is saved to <'filepath'>.  The public key portion of the
+    generated SPX key is saved to <'filepath'>.pub.  If the filepath is not
+    given, the KEYID is used as the filename and the keypair saved to the
+    current working directory.
+
+    The private key is encrypted according to 'cryptography's approach:
+    "Encrypt using the best available encryption for a given key's backend.
+    This is a curated encryption choice and the algorithm may change over
+    time."
+
+  <Arguments>
+    filepath:
+      The public and private key files are saved to <filepath>.pub and
+      <filepath>, respectively.  If the filepath is not given, the public and
+      private keys are saved to the current working directory as <KEYID>.pub
+      and <KEYID>.  KEYID is the generated key's KEYID.
+
+    password:
+      The password, or passphrase, to encrypt the private portion of the
+      generated SPX key.  A symmetric encryption key is derived from
+      'password', so it is not directly used.
+
+  <Exceptions>
+    securesystemslib.exceptions.FormatError, if the arguments are improperly
+    formatted.
+
+    securesystemslib.exceptions.CryptoError, if 'filepath' cannot be encrypted.
+
+  <Side Effects>
+    Writes key files to '<filepath>' and '<filepath>.pub'.
+
+  <Returns>
+    The 'filepath' of the written key.
+  """
+
+  # Generate a new SPX key object.
+  spx_key = securesystemslib.keys.generate_spx_key()
+
+  if not filepath:
+    filepath = os.path.join(os.getcwd(), spx_key['keyid'])
+
+  else:
+    logger.debug('The filepath has been specified.  Not using the key\'s'
+        ' KEYID as the default filepath.')
+
+  # Does 'filepath' have the correct format?
+  # Ensure the arguments have the appropriate number of objects and object
+  # types, and that all dict keys are properly named.
+  # Raise 'securesystemslib.exceptions.FormatError' if there is a mismatch.
+  securesystemslib.formats.PATH_SCHEMA.check_match(filepath)
+
+  # If the caller does not provide a password argument, prompt for one.
+  if password is None: # pragma: no cover
+
+    # It is safe to specify the full path of 'filepath' in the prompt and not
+    # worry about leaking sensitive information about the key's location.
+    # However, care should be taken when including the full path in exceptions
+    # and log files.
+    password = get_password('Enter a password for the SPX'
```

Comment:
Multi-line strings like this are kind of confusing to mentally parse when combined with string concatenation. Is there maybe a more compact way to write these?